lib/V8/v8-utils.cpp

```diff
@@ -1678,6 +1678,40 @@ static void JS_MakeAbsolute(v8::FunctionCallbackInfo<v8::Value> const& args) {
   TRI_V8_TRY_CATCH_END
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief was docuBlock JS_ChangeDirectoy
+////////////////////////////////////////////////////////////////////////////////
+
+static void JS_ChangeDirectoy(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  // extract arguments
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("changeDirectory(<path>)");
+  }
+
+  TRI_Utf8ValueNFC name(isolate, args[0]);
+
+  if (*name == nullptr) {
+    TRI_V8_THROW_TYPE_ERROR("<path> must be a string");
+  }
+
+  TRI_GET_GLOBALS();
+  V8SecurityFeature& v8security = v8g->_server.getFeature<V8SecurityFeature>();
+
+  if (!v8security.isAllowedToAccessPath(isolate, *name, FSAccessType::READ)) {
```

Comment:
I think changing the current directory should be forbidden in almost all cases, except when executed during testing. So let's make this more strict please

Reply:
moved to the spawning of external processes as a new parameter.